crates/topos/tests/config.rs

```diff
@@ -223,3 +257,96 @@ async fn test_command_init_precedence_cli_env() -> Result<(), Box<dyn std::error
 
     Ok(())
 }
+
+/// Test node up running from config file
+#[test_log::test(tokio::test)]
+async fn test_command_node_up() -> Result<(), Box<dyn std::error::Error>> {
+    let tmp_home_dir = tempdir()?;
+
+    // Create config file
+    let node_up_home_env = tmp_home_dir.path().to_str().unwrap();
+    let node_edge_path_env = polygon_edge_path(node_up_home_env).await;
+    let node_up_name_env = "TEST_NODE_UP";
+    let node_up_role_env = "full-node";
+    let node_up_subnet_env = "topos-up-env-subnet";
+
+    let mut cmd = Command::cargo_bin("topos")?;
+    cmd.arg("node")
+        .env("TOPOS_POLYGON_EDGE_BIN_PATH", &node_edge_path_env)
+        .env("TOPOS_HOME", node_up_home_env)
+        .env("TOPOS_NODE_NAME", node_up_name_env)
+        .env("TOPOS_NODE_ROLE", node_up_role_env)
+        .env("TOPOS_NODE_SUBNET", node_up_subnet_env)
+        .arg("init");
+
+    let output = cmd.assert().success();
+    let result: &str = std::str::from_utf8(&output.get_output().stdout)?;
+
+    // Run node init with cli flags
+    assert!(result.contains("Created node config file"));
+    let home = PathBuf::from(node_up_home_env);
+    // Verification: check that the config file was created
+    let config_path = home.join("node").join(node_up_name_env).join("config.toml");
+    assert!(config_path.exists());
+
+    // Generate polygon edge genesis file
+    let polygon_edge_bin = format!("{}/polygon-edge", node_edge_path_env);
+    generate_polygon_edge_genesis_file(
+        &polygon_edge_bin,
+        node_up_home_env,
+        node_up_name_env,
+        node_up_subnet_env,
+    )
+    .await?;
+    let polygon_edge_genesis_path = home
+        .join("subnet")
+        .join(node_up_subnet_env)
+        .join("genesis.json");
+    assert!(polygon_edge_genesis_path.exists());
+
+    // Run node up with using role and subnet from config file
+    let mut environment: Vec<(OsString, OsString)> = std::env::vars()
+        .map(|(key, value)| (key.into(), value.into()))
+        .collect();
+    environment.append(&mut vec![
+        (
+            "TOPOS_POLYGON_EDGE_BIN_PATH".into(),
+            (&node_edge_path_env).into(),
+        ),
+        ("TOPOS_HOME".into(), (&node_up_home_env).into()),
+        ("TOPOS_NODE_NAME".into(), (&node_up_name_env).into()),
+        ("RUST_LOG".into(), ("info").into()),
+    ]);
+
+    let mut child = subprocess::Popen::create(
+        &["cargo", "run", "--", "node", "up"],
+        subprocess::PopenConfig {
+            stdout: subprocess::Redirection::Pipe,
+            env: Some(environment),
+            ..Default::default()
+        },
+    )?;
+
+    // Use big timeout because `cargo run` sometimes builds before running
```

Comment:
This makes me a bit worried:
* This adds a significant time to our testing suite
* Makes the test a bit flaky because of `cargo run`

Reply:
I don't know - we have other tests with big timeout. @hadjiszs what do you think?